plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -21,33 +47,216 @@ public static GlobalKafkaConfiguration get() {
+        }
+    }
+
+    public FormValidation doTestBrokerConnection(@QueryParameter("brokerURL") final String brokerURL)
+            throws IOException, ServletException {
+        try {
+            String[] hostport = brokerURL.split(":");
+            String host = hostport[0];
+            int port = Integer.parseInt(hostport[1]);
+            testConnection(host, port);
+            return FormValidation.ok("Success");
+        } catch (Exception e) {
+            return FormValidation.error("Connection error : " + e.getMessage());
+        }
+    }
+
     @Override
-    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
-        this.connectionURL = json.getString("connectionURL");
-        this.zookeeperURL = json.getString("zookeeperURL");
+    public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
+        this.brokerURL = formData.getString("brokerURL");
+        this.zookeeperURL = formData.getString("zookeeperURL");
+        StandardUsernamePasswordCredentials credential = getCredential();
+        this.username = credential.getUsername();
+        this.password = credential.getPassword().getPlainText();
+        this.sslTruststoreLocation = formData.getString("sslTruststoreLocation");
+        this.sslTruststorePassword = formData.getString("sslTruststorePassword");
+        this.sslKeystoreLocation = formData.getString("sslKeystoreLocation");
+        this.sslKeystorePassword = formData.getString("sslKeystorePassword");
+        this.sslKeyPassword = formData.getString("sslKeyPassword");
         save();
-        return true;
+        return super.configure(req, formData);
+    }
+
+    private void testConnection(String host, int port) throws IOException {
+        new Socket(host, port);
+    }
+
+    private StandardUsernamePasswordCredentials getCredential() {
+        StandardUsernamePasswordCredentials credential = null;
+
+        List<StandardUsernamePasswordCredentials> credentials = CredentialsProvider.lookupCredentials(
+                StandardUsernamePasswordCredentials.class, Jenkins.get(), ACL.SYSTEM, Collections.emptyList());
+
+        IdMatcher matcher = new IdMatcher(credentialsId);
+        for (StandardUsernamePasswordCredentials c : credentials) {
+            if (matcher.matches(c)) {
+                credential = c;
+            }
+        }
+
+        return credential;
+    }
+
+    public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String credentialsId) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(credentialsId);
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ) && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
```

Comment:
Not required so far, because the global configuration must be only accessed by admins.